src/controllers/project.js

```diff
@@ -70,7 +72,100 @@ const getProjectSupporters = async (req, res, next) => {
   return next();
 };
 
+const createComment = async (req, res, next) => {
+  const { id } = req.params;
+  const { userID, content } = req.body;
+
+  try {
+    const project = await Project.findOne({ _id: id });
+
+    if (!project) return res.status(404).json({ message: 'Project not found' });
+
+    const comment = {
+      user: userID,
+      content,
+    };
+
+    project.comments.push(comment);
+
+    await project.save();
+
+    res.json({
+      message: 'Comment created successfully',
+      comments: project.comments,
+    });
+  } catch (error) {
+    res.status(422).json({ message: 'Unable to create comment' });
+  }
+  return next();
+};
+
+const isEmpty = (field) => !!field;
+
+const updateComment = async (req, res, next) => {
+  const { id, commentId } = req.params;
+  const { content } = req.body;
+
+  try {
+    const project = await Project.findById(id);
+
+    if (!project) return res.status(404).json({ message: 'Project not found' });
+
+    const comment = project.comments.find(
+      // eslint-disable-next-line no-underscore-dangle
+      (single) => ObjectId(single._id).toString() === commentId
+    );
+
+    if (!comment) return res.status(404).json({ message: 'Comment not found' });
+
+    comment.content = isEmpty(content) ? content : comment.content;
```

Comment:
here I did not get it you are asking if it is empty then update it?
and if not empty keep the original?
the condition confusing here
it should be !isEmpty

Reply:
Is empty function returns the boolean value of the field and here i am making ensure that i set the new value if exist in req body otherwise keep the old

Reply:
oh ok,
I thought if you did (!!) that means you want the opposite of the value like if it is false you add !! to make it true good work 👍

Reply:
no its to convert the value to boolean you can check it more here => https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_NOT#double_not_!!